src/Compilers/CSharp/Test/Emit/PDB/PortablePdbTests.cs

```diff
@@ -209,6 +210,151 @@ public static void Main()
+}
+public struct StructWithValue
+{
+    int PrivateData;
+}", options: TestOptions.DebugDll.WithCurrentLocalTime(referenceTimestamp));
+
+            string source = @"
+using System;
+
+class C
+{
+    public static void Main()
+    {
+        Console.WriteLine();
+    }
+}
+";
+            var originalCompilation = CreateCompilation(
+                Parse(source, "goo.cs"),
+                references: new MetadataReference[] { new CSharpCompilationReference(referenceCompilation) },
+                options: TestOptions.DebugDll.WithDeterministic(true));
+
+            var originalCompilationOptions = originalCompilation.Options;
+            var originalEmitOptions = EmitOptions.Default;
+
+            var peBlob = originalCompilation.EmitToArray(originalEmitOptions.
+                WithDebugInformationFormat(DebugInformationFormat.Embedded).
+                WithPdbChecksumAlgorithm(HashAlgorithmName.SHA384).
+                WithPdbFilePath(@"a/b/c/d.pdb"));
+
+            using (var peReader = new PEReader(peBlob))
+            {
+                var entries = peReader.ReadDebugDirectory();
+
+                AssertEx.Equal(new[] { DebugDirectoryEntryType.CodeView, DebugDirectoryEntryType.PdbChecksum, DebugDirectoryEntryType.Reproducible, DebugDirectoryEntryType.EmbeddedPortablePdb }, entries.Select(e => e.Type));
+
+                var codeView = entries[0];
+                var checksum = entries[1];
+                var reproducible = entries[2];
+                var embedded = entries[3];
+
+
+                using (var embeddedPdb = peReader.ReadEmbeddedPortablePdbDebugDirectoryData(embedded))
+                {
+                    var pdbReader = embeddedPdb.GetMetadataReader();
+
+                    var metadataReferenceBytes = getSingleBlob(PortableCustomDebugInfoKinds.MetadataReferenceInfo, pdbReader);
+                    var compilerFlagBytes = getSingleBlob(PortableCustomDebugInfoKinds.DeterministicCompilerFlags, pdbReader);
+
+                    Assert.NotEmpty(metadataReferenceBytes);
+                    Assert.NotEmpty(compilerFlagBytes);
+
+                    var (timestamp, fileSize, name, mvid) = parseMetadataReferenceInfo(metadataReferenceBytes);
+                    var compilerFlags = parseDeterministicCompilerFlags(compilerFlagBytes);
+
+                    // Check version
+                    Assert.Equal("", compilerFlags["compilerversion"]);
+
+                    // Check source encoding
+                    Assert.Equal("", compilerFlags["sourceencoding"]);
+
+                    // Check the metadata references
+                    Assert.Equal(referenceTimestamp.ToFileTime(), timestamp);
+                    Assert.Equal(500, fileSize);
+                    Assert.Equal(referenceCompilation.AssemblyName, name);
+                    Assert.Equal(Guid.NewGuid(), mvid);
+                }
+            }
+
+            static byte[] getSingleBlob(Guid infoGuid, MetadataReader pdbReader)
```

Comment:
>static  [](start = 12, length = 7)

Let's not use local functions here. You'll certainly need to write more than one test that uses these.

Reply:
Yea this is temporary. Some of these might need to be moved to another location entirely.